### ISSUE TITLE
appwrite function never gets completed

### ISSUE THREAD

**Issue by aadarshadhakalg** [appwrite/appwrite#1232]
## 🐛 Bug Report

When I try to execute a function that has no output and no errors, the function never gets completed. The timeout was set to 15 seconds. But, I guess the timeout feature is not working properly.

## Have you spent some time to check if this issue has been raised before?

yes

## To Reproduce

```
void main(){
}
```
on Dart 2.10 runtime.

This is the function I tried to run.

## Expected behavior

The function should output nothing and finish executing. 

## Actual Behavior

Function status always waiting and sometimes it also blocks other executions afterwards.

**Comment by doricci** [comment]
I have the same issue with nodejs or de o environnement.
Il the logs of appwrite-worker-functions the ñame of a containers is already used and The function crash.

**Comment by alexweininger** [comment]
I have experienced the same thing. These are indefinitely "waiting".
<img width="928" alt="Screen Shot 2021-06-01 at 5 00 02 PM" src="https://user-images.githubusercontent.com/12476526/120395584-d79dfd00-c2fa-11eb-9a9d-3e1cff10b76c.png">

**Comment by doricci** [comment]
Here is some stacktrace of the container `appwrite-worker-functions` after facing the issue again this morning
```bash
| Itterating function: Account Created
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | Triggered function: account.create
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | [critical] (Job{v1-functions} | ID: 6a2a3518feabd2c3e4df90093ac51e5e | FunctionsV1 | [{"projectId":"60b1ebdbd07ef","webhooks":[],"userId":"","event":"account.create","eventData":{"$id":"60b7076581ce1","name":"d1","registration":1622607717,"status":0,"passwordUpdate":1622607717,"email":"MY_EMAIL@gmail.com","emailVerification":false,"prefs":[]},"functionId":null,"executionId":null,"trigger":"event"}]) has failed Exception: Failed to create function environment: docker: Error response from daemon: Conflict. The container name "/appwrite-function-60b4edafd3d08" is already in use by container "b403061b82dc03954faeb4471674b18f401fe7e90783d43fb33748b8b400d896". You have to remove (or rename) that container to be able to reuse that name.
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | See 'docker run --help'.
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    |  in /usr/src/code/app/workers/functions.php:432
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | Stack trace:
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | #0 /usr/src/code/app/workers/functions.php(200): FunctionsV1->execute('event', '60b1ebdbd07ef', '', Object(Appwrite\Database\Database), Object(Appwrite\Database\Document), 'account.create', '{"$id":"60b7076...', '', Array, '', '')
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | #1 /usr/src/code/vendor/resque/php-resque/lib/Resque/Job.php(201): FunctionsV1->perform()
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | #2 /usr/src/code/vendor/resque/php-resque/lib/Resque/Worker.php(303): Resque_Job->perform()
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | #3 /usr/src/code/vendor/resque/php-resque/lib/Resque/Worker.php(242): Resque_Worker->perform(Object(Resque_Job))
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | #4 /usr/src/code/vendor/resque/php-resque/bin/resque(185): Resque_Worker->work(5, false)
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | #5 {main}
appwrite_appwrite-worker-functions.1.nrukuniwgkdv@rpi4-node1    | [notice] Starting work on (Job{v1-functions} | ID: 9e73facbe3115c55572ae7e03f7d1e47 | FunctionsV1 | [{"projectId":"60b1ebdbd07ef","webhooks":[],"userId":"","event":"account.sessions.create","eventData":{"$id":"60b70765cd5d7","userId":"60b7076581ce1","expire":1654143717,"provider":"email","providerUid":"MY_EMAIL@gmail.com","providerToken":"","ip":"10.0.0.90","osCode":"LIN","osName":"GNU\/Linux","osVersion":"","clientType":"browser","clientCode":"CH","clientName":"Chrome","clientVersion":"90.0","clientEngine":"Blink","clientEngineVersion":"","deviceName":"desktop","deviceBrand":"","deviceModel":"","countryCode":"--","countryName":"Inconnu","current":true},"functionId":null,"executionId":null,"trigger":"event"}])
```

**Comment by doricci** [comment]
It's the same bug as the bug #1100 I think

**Comment by TorstenDittmann** [comment]
Thanks for the report!

I will investigate this issue to see how we can fix it 🙂 

Thanks alot for the cooperation :bow:

**Comment by didair** [comment]
I am also seeing this issue 1/3 of the times. If you need help replicating the error and checking logs etc. You can contact me on discord (andrew)

**Comment by eldadfux** [comment]
@TorstenDittmann I would guess the functions worker is for some reason not getting the list of running containers properly, not sure what might help us to reproduce this situation.

**Comment by Meldiron** [comment]
![image](https://user-images.githubusercontent.com/19310830/121750498-caf48280-cb0c-11eb-9bb3-d952f60df34f.png)

Same problem here. It starts multiple `appwrite/env-node-15.5:1.0.0` containers (10+, probably one for every frozen execution). I have no additional info as this happened randomly and cannot reproduce it. I will let you know if I find some system in this behaviour.

**Comment by eldadfux** [comment]
Good news. I created a PR on #1288 to fix the problem that caused the creation of already running functions which was causing a docker conflict error. This error was happening when the max container limit has reached and the cleanup method was wrongly removing the containers list array keys.

Until the fix is available a workaround can be to increase the number of allowed containers and recreate the `appwrite-worker-functions` container.

**Comment by eldadfux** [comment]
This is now fixed and available as part of Appwrite 0.9:
https://dev.to/appwrite/announcing-appwrite-0-9-the-open-source-firebase-alternative-53ho 🥳 🎉
